### ISSUE TITLE
error while loading nd2 image

### ISSUE THREAD

**Issue by jboulanger** [tlambert03/nd2#91]
* nd2 version: 0.4.0 
* Python version: 3.9 / 3.10
* Operating System: ubuntu 18.04 / win 10

### Description
Loading a nd2 image leads to corrupted data. Rows are shifted by 1 pixels and there is a residual error even after shifing rows back compared to the image decoded by bioformats and a tif exported from NIS Elements.

Data accessible from here: https://cloud.mrc-lmb.cam.ac.uk/s/jJMMtqAaRALnnTq

### What I Did

```
import nd2
import matplotlib.pyplot as plt
import numpy as np
import javabridge
import bioformats 
import tifffile
javabridge.start_vm(class_path=bioformats.JARS,run_headless=True)
fname = 'test.nd2'
imgs = [
    nd2.imread(fname),
    np.stack([line[np.arange(k,k+line.shape[0])%line.shape[0]] for k,line in enumerate(nd2.imread(fname))]),
    bioformats.load_image(fname, rescale=False),
    tifffile.imread(fname.replace('.nd2','.tif'))]
fig, ax = plt.subplots(1,len(imgs),figsize=(20,40))
titles = ['nd','nd2 shifted','bioformats','tiff']
for img,a,t in zip(imgs,ax,titles):    
    a.imshow(img)
    a.axis('off')
    a.set_title(t)

[np.abs(img.astype(float)-imgs[3].astype(float)).max() for img in imgs]

```

![image](https://user-images.githubusercontent.com/3415561/189355506-6eb606f0-583a-4cc4-acfc-dfe0fce654d4.png)

Max absolute error with tiff reference: [2707.0, 1564.0, 0.0, 0.0]
```
plt.imshow(np.abs(imgs[1].astype(np.float32)-imgs[3].astype(np.float32)),vmax=100)
plt.colorbar()
```
![image](https://user-images.githubusercontent.com/3415561/189363435-8c5d43fd-7285-4ab3-8ed5-003b3b8ef925.png)

**Comment by tlambert03** [comment]
thanks for the issue and the data @jboulanger.  I think you're running into the same thing as #88 ... which was just fixed in #90

I'm not sure why nd2 sometimes has internally inconsistent metadata (where the widthPx is sometimes _not_ a clean multiple of widthBytes * bytesPerPixel)... but your dataset appears to be one of them.

On the main branch, with #90 merged, `nd2.imread` appears to work fine for your file:

<img width="740" alt="Screen Shot 2022-09-09 at 10 38 51 AM" src="https://user-images.githubusercontent.com/1609449/189376352-cb6167c4-b614-4cec-a04a-a34ca4089f30.png">

I'll cut a new release soon

**Comment by jboulanger** [comment]
thanks

**Comment by tlambert03** [comment]
releasing now... will take 30-60 mins to be available on pip.
Closing this "optimistically" ... but feel free to reopen if the new version doesn't fix your issue

**Comment by jboulanger** [comment]
Just tested again on the image, the original image (tif or loaded with bioformat) had a width of 4801 but the one opened with the new version has an extra column at the end (width=4802).

**Comment by tlambert03** [comment]
Yeah, I can add something to crop that I suppose.  I think that at this point, it becomes a bit of a matter of choice with how to deal with the fact that the image metadata _itself_ says that the data on disk is 4802 pixels.  Let me try to learn a bit more from LIM (the company that created the nd2 format) about these relatively rare cases where the metadata is internally inconsistent, to see how they would "officially" have it be interpreted (rather than just mimicking bioformats choice, which may also just be one person's decision).  thanks for letting me know!

**Comment by tlambert03** [comment]
Yeah thanks, and the nis viewer also opens it at 4801

**Comment by tlambert03** [comment]
merged a better fix in #92, and it will be in v0.4.2 which is building now

**Comment by jboulanger** [comment]
Amazing!
